core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for rendering learner action HTML strings.
+ */
+
+oppia.factory('LearnerActionRenderService', [
+  'ACTION_TYPE_ANSWER_SUBMIT', 'ACTION_TYPE_EXPLORATION_START',
+  'ACTION_TYPE_EXPLORATION_QUIT', 'ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS',
+  function(
+      ACTION_TYPE_ANSWER_SUBMIT, ACTION_TYPE_EXPLORATION_START,
+      ACTION_TYPE_EXPLORATION_QUIT, ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS) {
+    var renderExplorationStartActionHTML = function(stateName) {
+      var htmlString =
+        'Started exploration at card "' + stateName + '".<br><br>';
+      return htmlString;
+    };
+
+    var renderExplorationQuitActionHTML = function(
+        stateName, timeSpentInStateSecs) {
+      var htmlString =
+        'Left the exploration after spending a total of ' +
+        timeSpentInStateSecs + ' seconds on card "' + stateName +
+        '".<br><br>';
+      return htmlString;
+    };
+
+    var renderContinueButtonSubmitActionHTML = function(
+        stateName, timeSpentInStateSecs) {
+      var htmlString =
+        'Pressed "Continue" to move to card "' + stateName + '" after ' +
+        timeSpentInStateSecs + ' seconds.<br><br>';
+      return htmlString;
+    };
+
+    var renderAnswerSubmitActionHTML = function(
+        answer, destStateName, timeSpentInStateSecs, currentStateName) {
+      var htmlString;
+      if (currentStateName === destStateName) {
+        htmlString =
+          'Submitted answer "' + answer + '" in card "' + currentStateName +
+          '".<br><br>';
+      } else {
+        htmlString =
+          'Submitted answer "' + answer + '" and moved to card "' +
+          destStateName + '" after spending ' + timeSpentInStateSecs +
+          ' seconds on card "' + currentStateName + '".<br><br>';
+      }
+      return htmlString;
+    };
+
+    var renderLearnerActionsTableForMultipleIncorrectIssue = function(
+        finalBlock) {
+      var index = finalBlock.length - 1;
+      var stateName =
+        finalBlock[index].actionCustomizationArgs.state_name.value;
+
+      var tableHTML = '<table><tr><th>Answer</th><th>Feedback</th></tr>';
```

Comment:
Add a class to this table so we can style it with CSS: `<table class="learner-actions-table">`.

Reply:
Done.